specification/securitycenter/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json

```diff
@@ -0,0 +1,1955 @@
+                        },
+                        "provisioningState": {
+                            "type": "string",
+                            "description": "Gets the provisioning state of the Just-in-Time policy.",
+                            "enum": [
+                                "Succeeded",
+                                "Failed"
+                            ]
+                        }
+                    }
+                }
+            },
+            "allOf": [{
+                    "$ref": "#/definitions/Resource"
+                },
+                {
+                    "$ref": "#/definitions/Kind"
+                },
+                {
+                    "$ref": "#/definitions/Location"
+                }
+            ]
+        },
+        "JitNetworkAccessPolicyVirtualMachine": {
+            "type": "object",
+            "required": [
+                "id",
+                "ports"
+            ],
+            "properties": {
+                "id": {
+                    "type": "string",
+                    "description": "Resource ID of the virtual machine that is linked to this policy"
+                },
+                "ports": {
+                    "type": "array",
+                    "description": "Port configurations for the virtual machine",
+                    "items": {
+                        "required": [
+                            "allowedSourceAddressPrefix",
+                            "maxRequestAccessDuration",
+                            "number",
+                            "protocol"
+                        ],
+                        "type": "object",
+                        "properties": {
+                            "number": {
+                                "$ref": "#/definitions/PortNumber"
+                            },
+                            "protocol": {
+                                "type": "string",
+                                "enum": [
+                                    "TCP",
+                                    "UDP",
+                                    "*"
+                                ]
+                            },
+                            "allowedSourceAddressPrefix": {
+                                "type": "string"
+                            },
+                            "maxRequestAccessDuration": {
+                                "type": "string",
+                                "description": "Maximum duration requests can be made for. In ISO 8601 duration format. Minimum 5 minutes, maximum 1 day"
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "JitNetworkAccessRequest": {
+            "type": "object",
+            "required": [
+                "requestor",
+                "startTimeUtc",
+                "virtualMachines"
+            ],
+            "properties": {
+                "virtualMachines": {

```

Comment:
When you define a complex object inline, you forget to add `"type": "object"`. Not specifying the `"type"` can have a different meaning for that model in the swagger spec. Please read [this doc](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#understanding-the-importance-of-type-keyword-while-definining-model-types) for understanding the importance of "type" for a complex object. 
Thus defining complex objects separately makes it cleaner to read and it becomes easy to catch errors.